packages/components/icon/src/Icon.variants.tsx

```diff
@@ -3,7 +3,18 @@ import { cva, VariantProps } from 'class-variance-authority'
 
 export const iconVariants = cva(['fill-current'], {
   variants: {
+    intent: makeVariants<'intent'>({
+      current: ['text-current'],
+      primary: ['text-primary'],
+      secondary: ['text-secondary'],
+      success: ['text-success'],
+      alert: ['text-alert'],
+      error: ['text-error'],
+      info: ['text-info'],
+      neutral: ['text-neutral'],
+    }),
     size: makeVariants<'size'>({
+      current: ['w-current', 'h-current'],
```

Comment:
I chose the name `current` instead of `auto` because it will be confused with the auto css value